File: frontend/src/app/model/figure/figureOperation.ts
import { Figure } from './figure';

export type FigureOperationType =
  | 'NEW_OPERATION'
  | 'MOVEMENT_OPERATION'
  | 'ROTATION_OPERATION'
  | undefined;

export interface FigureOperation {
  figure: Figure;
  initialTimeStamp: number;
  timeStamp: number;
  operation: FigureOperationType;
}

File: api/src/main/java/org/jmedina/jtetris/api/model/FigureOperation.java
package org.jmedina.jtetris.api.model;

import org.jmedina.jtetris.api.enumeration.FigureOperationEnumeration;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Jorge Medina
 *
 */
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class FigureOperation {

	private FigureOperationEnumeration operation;
	private Figure figure;
	private long initialTimeStamp;
	private long timeStamp;

}
